### PR TITLE
Restructure and rewrite of prereqs content

### PR DESCRIPTION
**Pull request by danmacpherson**
Brief summary of proposed changes:
- Created a new Planning section. This is to give users some high level guidance on how to plan the overall structure of their environment and their methods of installation.
- Revised content in the Prereqs section to clarify certain parts
- Separated Host Preparation into its own section on the TOC. This is to distinguish the requirements from the host prep procedures.
- Renamed "RPM vs Containerized" to "Containerized Services". The former sets up the expectation that it will discuss differences between RPM and Containerized installation; however, the content does not reflect this.

Preview of the content here: http://file.bne.redhat.com/~dmacpher/OpenShift/openshift-enterprise/master/install_config/

### CODE REVIEW

**Comment by danmacpherson** [comment]
@vikram-redhat @adellape @bfallonf @tpoitras @aheslin @ahardin-rh -- Tagging you all for peer review.

I've discussed with Alex the plan for #2741, which is to commit this PR and Alex can rebase off my changes.

File: install_config/install/planning.adoc [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+toc::[]
+
+[[inital-planning]]
+== Initial Planning
+
+For production environments, several factors influence installation. Consider
+the following questions as you read through the documentation:
+
+* _Which installation method do you want to use?_ The xref:environment-scenarios[Installation Methods]
```
Comment (adellape) [inline install_config/install/planning.adoc]: s/environment-scenarios/installation-methods/

File: install_config/install/planning.adoc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+* _How many hosts do you require in the cluster?_ The xref:environment-scenarios[Environment Scenarios]
+section provides multiple examples of Single Master and Multiple Master
+configurations.
+
+* _How many pods are required in your cluster?_ The xref:sizing[Sizing Considerations]
+section provides limits for nodes and pods so you can calculate how large your
+environment needs to be.
+
+* _Is xref:../../admin_guide/high_availability.adoc#admin-guide-high-availability[high availability]
```
Comment (adellape) [inline install_config/install/planning.adoc]: If this is meaning to refer to HA masters, then it probably should link to:

../../architecture/infrastructure_components/kubernetes_infrastructure.adoc#high-availability-masters

The current admin_guide topic it's linking to is about HA services.

File: install_config/install/planning.adoc [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+required?_ High availability is recommended for fault tolerance. In this
+situation, you might aim to use the xref:multi-masters-using-native-ha][Multiple Masters Using Native HA]
```
Comment (adellape) [inline install_config/install/planning.adoc]: Stray `]`

File: install_config/install/rpm_vs_containerized.adoc [hunk 4]
```diff
@@ -13,31 +13,23 @@ toc::[]
...
-endif::[]
-uses RPMs. Alternatively, you can use the containerized method, which deploys
-containerized {product-title} master and node components. When targeting a RHEL
-Atomic Host system, the containerized method is the only available option, and
-is automatically selected for you based on the detection of the
-*_/run/ostree-booted_* file.
-
-You can easily deploy environments mixing containerized and RPM based
-installations. For the
+This section explores some of the preparation required to install {product-name}
```
Comment (adellape) [inline install_config/install/rpm_vs_containerized.adoc]: s/{product-name}/{product-title}/

File: install_config/install/rpm_vs_containerized.adoc [hunk 5]
```diff
@@ -48,8 +40,8 @@ Installation] for load balancer requirements using the native HA method.
 ====
 endif::[]
 
-The following sections detail the differences between the RPM and containerized
-methods.
+The following sections detail the preparation for a containerized {product-name}
```
Comment (adellape) [inline install_config/install/rpm_vs_containerized.adoc]: s/{product-name}/{product-title}/

File: _build_cfg.yml [hunk 6]
```diff
@@ -222,11 +222,13 @@ Topics:
...
-      - Name: Overview
-        File: index
+      - Name: Planning
+        File: planning
       - Name: Prerequisites
         File: prerequisites
-      - Name: RPM vs Containerized
+      - Name: Host Preparation
+        File: host_preparation
+      - Name: Containerized Services
```
Comment (adellape) [inline _build_cfg.yml]: I know the text in the topic says "install OCP as a set of services within containers", but I think "Services" is too broad a word here (plus "service" is an overloaded word in OpenShift). How about:

Containerized Components
Containerized Infrastructure

....mmm, that's all I got right now.

File: install_config/install/planning.adoc (hunk 1 above)
Comment (danmacpherson) [inline install_config/install/planning.adoc]: Corrected.

File: install_config/install/planning.adoc (hunk 2 above)
Comment (danmacpherson) [inline install_config/install/planning.adoc]: I've left this link intact from the original content. However, we can change this if necessary.

File: install_config/install/planning.adoc (hunk 3 above)
Comment (danmacpherson) [inline install_config/install/planning.adoc]: Removed.

File: install_config/install/rpm_vs_containerized.adoc (hunk 4 above)
Comment (danmacpherson) [inline install_config/install/rpm_vs_containerized.adoc]: Fixed.

File: install_config/install/rpm_vs_containerized.adoc (hunk 5 above)
Comment (danmacpherson) [inline install_config/install/rpm_vs_containerized.adoc]: Fixed.

File: _build_cfg.yml (hunk 6 above)
Comment (danmacpherson) [inline _build_cfg.yml]: I like Containerized Components, so I'll change it to that. My main concern was moving away from the title "RPM vs Containerized", which isn't an accurate title description. (Though I've added some proper RPM vs Containered content in the Planning section)

**Comment by bfallonf** [comment]
LGTM

**Comment by vikram-redhat** [comment]
Ok, so if I understand it correctly, we need to close this PR as @danmacpherson 's commits are now rebased into #2776. So, I am going to go ahead and close it. @danmacpherson  and @adellape, please correct me if I am wrong and reopen with comments.

**Comment by danmacpherson** [comment]
@Vikram, I think that might be the case though I'm not sure. I thought the plan was to merge my commits, then rebase. But it looks like @adellape has already rebased them and will be including them in #2776.

Not sure what the plan is now, but I'm fine with whatever method is chosen.

**Pull request closed without merging**